inc/class-capsule-client.php

```diff
@@ -0,0 +1,1460 @@
+				}
+				die();
+
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Process server info update request (on AJAX).
+	 *
+	 * @param array $args Request args.
+	 * @return void
+	 */
+	private function process_update_server_info( $args ) {
+		$errors = array();
+		if ( empty( $args['nonce'] ) || ! wp_verify_nonce( $args['nonce'], '_cap_client_server_management' ) ) {
+			wp_send_json( array(
+				'result' => 'error',
+				'errors' => array(
+					array(
+						'message' => esc_html__( 'Invalid request.', 'capsule' ),
+						'type'    => 'general',
+					),
+				),
+				'data'   => array(
+					'name'    => $args['server_name'],
+					'api_key' => $args['api_key'],
+					'url'     => $args['server_url'],
+				),
+			));
+		}
+		$server_id = (int) $args['server_id'];
+		if ( $server_id > 0 ) {
+			$test_errors       = $this->test_credentials( $args['api_key'], $args['server_url'] );
+			$duplicate_errors  = $this->duplicate_server_check( $args['server_name'], $args['server_url'], $server_id );
+			$validation_errors = array_merge( $test_errors, $duplicate_errors );
+
+			if ( empty( $validation_errors ) ) {
+				$server = $this->update_server( $server_id, $args );
+				if ( $server ) {
+					wp_send_json( array(
+						'data'   => array(
+							'name'    => $server->post_title,
+							'api_key' => $server->api_key,
+							'url'     => $server->url,
+						),
+						'result' => 'success',
+					));
+				} else {
+					$errors[] = array(
+						'message' => esc_html__( 'Could not save server.', 'capsule' ),
+						'type'    => 'general',
+					);
+				}
+			} else {
+				$errors = $validation_errors;
+			}
+		} else {
+			$errors[] = array(
+				'message' => esc_html__( 'No server ID found.', 'capsule' ),
+				'type'    => 'general',
+			);
+		}
+
+		wp_send_json( array(
+			'result' => 'error',
+			'errors' => $errors,
+			'data'   => array(
+				'name'    => $args['server_name'],
+				'api_key' => $args['api_key'],
+				'url'     => $args['server_url'],
+			),
+		));
+	}
+
+	/**
+	 * Return all terms in a group of taxonomies
+	 *
+	 * @param string|array $taxonomies Taxonomy name or list of Taxonomy names.
+	 * @return array                   Array of term objects.
+	 **/
+	private function get_taxonomy_terms( $taxonomies ) {
+		return get_terms( $taxonomies, array(
+			'hide_empty' => false,
+			'orderby'    => 'slug',
+			'order'      => 'ASC',
+		));
+	}
+
+	/**
+	 * Get a list of servers
+	 * Returns an array of servers with the key as the post type slug:
+	 * array(
+	 *    'server-1' => array(
+	 *        'api-key' => '12345abc',
+	 *        'url' => 'http://capsule-server-1.com';
+	 *    ),
+	 *    'server-2' => array(
+	 *        'api-key' => 'A4d*DYnohiO',
+	 *        'url' => 'http://capsule-server-2.com';
+	 *    )
+	 * )
+	 *
+	 * Potential feature - Filter servers based on owner for multi-user support
+	 **/
+	public function get_servers() {
+		$query = new WP_Query( array(
+			'post_type'      => 'server',
+			'posts_per_page' => -1,
```

Comment:
Maybe it's better if we make this with a more realistic limit then using an unlimited query? Like 500?